src/fmu/dataio/_objectdata_provider.py

```diff
@@ -477,9 +535,10 @@ def _derive_timedata(self):
             return {}
 
         if self.dataio.legacy_time_format:
-            return self._derive_timedata_legacy()
+            timedata = self._derive_timedata_legacy()
         else:
-            return self._derive_timedata_newformat()
+            timedata = self._derive_timedata_newformat()
+        return timedata
```

Comment:
Why extra code here?

Reply:
Sorry, came up in the linting process, couldn't resist.
_objectdata_provider.py:479:8: R1705: Unnecessary "else" after "return", remove the "else" and de-indent the code inside it (no-else-return) (pylint 2.15.5): But no skin off my nose to remove it if you that is what you want)

Reply:
ok, not a big deal